typed-racket-lib/typed-racket/base-env/prims-contract.rkt

```diff
@@ -250,10 +251,17 @@
 ;;  make-predicate
 ;;  cast
 
-;; Helper to construct syntax for contract definitions
+;; Helpers to construct syntax for contract definitions
+;; make-contract-def-rhs : Type-Stx Boolean Boolean -> Syntax
 (define (make-contract-def-rhs type flat? maker?)
   (contract-def-property #'#f `#s(contract-def ,type ,flat? ,maker? untyped)))
 
+;; make-contract-def-rhs/from-typed : Symbol Boolean Boolean -> Syntax
+(define (make-contract-def-rhs/from-typed sym flat? maker?)
```

Comment:
Also, can we just delay all `contract-def-property` constructions so that we have a uniform interface?